draft-ietf-tls-esni.md

```diff
@@ -577,13 +577,13 @@ authenticated, the client MUST abort the connection with an "ech_required"
 alert. It then processes the "retry_configs" field from the server's
 "encrypted_client_hello" extension.
 
-If one of the values contains a version supported by the client, it can regard
-the ECH keys as securely replaced by the server. It SHOULD retry the handshake
-with a new transport connection, using that value to encrypt the ClientHello.
-The value may only be applied to the retry connection. The client MUST continue
-to use the previously-advertised keys for subsequent connections. This avoids
-introducing pinning concerns or a tracking vector, should a malicious server
-present client-specific retry keys to identify clients.
+If at least one of the values contains a version supported by the client, it can
+regard the ECH keys as securely replaced by the server. It SHOULD retry the
+handshake with a new transport connection, using that value to encrypt the
```

Comment:
"at least one of the values" doesn't work with the use of "that value" later on in the paragraph. The wording should probably align with the first paragraph of {{send-ech}}.

Reply:
Updated the text here to clarify that the client is meant to retry with retry configurations supplied by the server. I also updated the first paragraph of {{send-ech}} to clarify that a "supported configuration" is one for which `ECHConfig.version` indicates a version of ECH known to the client.